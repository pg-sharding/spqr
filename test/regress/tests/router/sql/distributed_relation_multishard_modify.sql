\c spqr-console
CREATE DISTRIBUTION ds1 COLUMN TYPES int;
CREATE KEY RANGE krid1 FROM 1 ROUTE TO sh1 FOR DISTRIBUTION ds1;
CREATE KEY RANGE krid2 FROM 30 ROUTE TO sh2 FOR DISTRIBUTION ds1;
ALTER DISTRIBUTION ds1 ATTACH RELATION distrr_mm_test DISTRIBUTION KEY id;

\c regress
CREATE TABLE distrr_mm_test (id INTEGER, t TEXT);

COPY distrr_mm_test(id,t) FROM STDIN;
1	'u'
2	'u'
3	'u'
4	'u'
5	'u'
\.

COPY distrr_mm_test(id,t) FROM STDIN;
31	'u'
32	'u'
33	'u'
34	'u'
35	'u'
\.


UPDATE distrr_mm_test SET t = 'm' WHERE id IN (3, 34) /* __spqr__engine_v2: true */;


SELECT * FROM distrr_mm_test ORDER BY id, t /*__spqr__execute_on: sh1 */;
SELECT * FROM distrr_mm_test ORDER BY id, t /*__spqr__execute_on: sh2 */;


DROP TABLE distrr_mm_test;

\c spqr-console
DROP DISTRIBUTION ALL CASCADE;
DROP KEY RANGE ALL;
