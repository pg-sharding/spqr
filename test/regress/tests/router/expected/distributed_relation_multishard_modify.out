\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

CREATE DISTRIBUTION ds1 COLUMN TYPES int;
    add distribution    
------------------------
 distribution id -> ds1
(1 row)

CREATE KEY RANGE krid1 FROM 1 ROUTE TO sh1 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 1
(1 row)

CREATE KEY RANGE krid2 FROM 30 ROUTE TO sh2 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 30
(1 row)

ALTER DISTRIBUTION ds1 ATTACH RELATION distrr_mm_test DISTRIBUTION KEY id;
           attach table            
-----------------------------------
 relation name   -> distrr_mm_test
 distribution id -> ds1
(2 rows)

\c regress
CREATE TABLE distrr_mm_test (id INTEGER, t TEXT);
NOTICE: send query to shard(s) : sh1,sh2
COPY distrr_mm_test(id,t) FROM STDIN;
NOTICE: send query to shard(s) : sh1,sh2
COPY distrr_mm_test(id,t) FROM STDIN;
NOTICE: send query to shard(s) : sh1,sh2
UPDATE distrr_mm_test SET t = 'm' WHERE id IN (3, 34) /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh1,sh2
SELECT * FROM distrr_mm_test ORDER BY id, t /*__spqr__execute_on: sh1 */;
NOTICE: send query to shard(s) : sh1
 id |  t  
----+-----
  1 | 'u'
  2 | 'u'
  3 | m
  4 | 'u'
  5 | 'u'
(5 rows)

SELECT * FROM distrr_mm_test ORDER BY id, t /*__spqr__execute_on: sh2 */;
NOTICE: send query to shard(s) : sh2
 id |  t  
----+-----
 31 | 'u'
 32 | 'u'
 33 | 'u'
 34 | m
 35 | 'u'
(5 rows)

DROP TABLE distrr_mm_test;
NOTICE: send query to shard(s) : sh1,sh2
\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

DROP DISTRIBUTION ALL CASCADE;
   drop distribution    
------------------------
 distribution id -> ds1
(1 row)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

